.github/workflows/build-and-test.yml

```diff
@@ -3,28 +3,43 @@ name: build-and-test
 on:
   push:
     # all branches
+    paths-ignore:
+      - 'documentation/**'
   pull_request:
     branches:
       - main
       - master
+    paths-ignore:
+      - 'documentation/**'
 
   # This enables the Run Workflow button on the Actions tab.
   workflow_dispatch:
 
 jobs:
   build-and-test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
Does this need to be a matrix if it is only running on a single platform and no other variation?

Reply:
I leave it ready in case we want to add Mac OS later.

Reply:
Should it be deleted?

Reply:
It doesn't make much difference to me.  I guess in the long run it's good if we run most automation on both Linux and macOS.

Reply:
Ok, then I merge.